test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -1823,6 +1823,7 @@ protected static boolean isXPackTemplate(String name) {
             case "logs":
             case "logs-settings":
             case "logs-mappings":
+            case "logs@custom":
```

Comment:
This is required, otherwise `ESRestTestCase` fails during [`cleanUpCluster()`](https://github.com/elastic/elasticsearch/blob/97d1ca05958cba0b24cd55b2367ab569feaab226/test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java#L398) - if first issues a warning when trying to [delete index templates](https://github.com/elastic/elasticsearch/blob/97d1ca05958cba0b24cd55b2367ab569feaab226/test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java#L731) and then[ fails the test](https://github.com/elastic/elasticsearch/blob/97d1ca05958cba0b24cd55b2367ab569feaab226/test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java#L842-L845) because it finds an unexpected template.

Reply:
We'll need to delete the `logs@custom` template, otherwise it'll leak into other test cases. I think the issue is that even though you've set `"ignore_missing_component_templates": ["logs@custom"]`, it's not possible to delete this component template as ES thinks it's required.

This sounds like a bug in the delete component template API or an oversight when adding the `ignore_missing_component_templates` option in https://github.com/elastic/elasticsearch/pull/92436.

I suppose what's missing is that `org.elasticsearch.cluster.metadata.ComposableIndexTemplate#getIgnoreMissingComponentTemplates` is considered here:

https://github.com/elastic/elasticsearch/blob/97d1ca05958cba0b24cd55b2367ab569feaab226/server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java#L458-L465

Reply:
Hopefully this leakage to other tests is what causes all these CI tests, although I can't really see how- it is just a component template that was register to the cluster. The failures actually may be related to something else that was overlooked when `ignore_missing_component_templates`, but I can't say what yet...

> I think the issue is that even though you've set `"ignore_missing_component_templates": ["logs@custom"]`, it's not possible to delete this component template as ES thinks it's required

The message is: 
```
component templates [logs@custom] cannot be removed as they are still in use by index templates [logs]
```
maybe this is what it means. 

Let me try that and remove the `logs@custom` template manually at the end of the test

Reply:
It's not possible to delete a component template that's "in use", even manually. The issue is that the current definition of "in use" is very strict and doesn't consider `ignore_missing_component_templates`.

Reply:
I meant that I will apply the change you propose AND delete manually at the end of the test, but this is not required as once I remove it from the list of the inherent component template list, it should be deleted automatically.
Should be fixed with c6ee343407d52043268e188e05e97854f709ff2a.

Other tests are still failing, I will keep looking into that

Reply:
Right. There was another related bug. Currently this PR contains a fix for both, do you think it is a proper fix? 
I will open a separate PR soon to address both, as @felixbarny [proposed](https://github.com/elastic/elasticsearch/pull/95481#pullrequestreview-1397400697), I am finding my way to add proper tests for both cases.

Reply:
I opened https://github.com/elastic/elasticsearch/pull/95527 to fix the related bugs